### ISSUE TITLE
Parallelisation for shared libraries

### ISSUE THREAD

**Issue by miguelmazetto** [CompArchCam/Janus#2]
Hi, im trying to implement Janus on a Garrys Mod server, a game that uses very old libraries being most of them single-threaded (like physics engine). However i noticed that Janus doesnt parallelize the libraries that were loaded at runtime, can that be solved?

**Comment by timothymjones** [comment]
Yes, that should be possible.  We have this capability (analysing and modifying dynamically loaded libraries) in experimental form for another project.  Although it's not a parallelisation project, I don't see why it couldn't apply here.  I'll find out its status and get back to you.

**Comment by timothymjones** [comment]
It's been an extremely long time since this issue was opened and apologies for this.  Today I merged in a security branch that provides the ability to do security analysis on all code, both statically found and dynamically generated.  Shared libraries can be analysed individually and their rewrite schedules loaded when they are.  Although this is for security, not parallelisation, I hope that if you still need this functionality then you can get it from this branch, or use this as inspiration.
